driver/sparql/Connection.php

```diff
@@ -274,6 +276,118 @@ protected function getStatement(): StatementStub {
     return $this->statementStub;
   }
 
+  /**
+   * Creates an array of database connection options from a URL.
+   *
+   * @param string $url
+   *   The URL.
+   * @param string $root
+   *   The root directory of the Drupal installation. Some database drivers,
+   *   like for example SQLite, need this information.
+   *
+   * @return array
+   *   The connection options.
+   *
+   * @throws \InvalidArgumentException
+   *   Exception thrown when the provided URL does not meet the minimum
+   *   requirements.
+   *
+   * @see \Drupal\Core\Database\Database::convertDbUrlToConnectionInfo()
+   *
+   * @internal
+   *   This method should not be called. Use
+   *   \Drupal\Core\Database\Database::convertDbUrlToConnectionInfo() instead.
+   */
+  public static function createConnectionOptionsFromUrl($url, $root) {
```

Comment:
Then why do we have these methods here if there is no need and there are already some in the core database class? Because if there is some difference between here and the core database's method, then the `@internal` is not needed.

Reply:
This copied from the core connection abstract as we don't extend that class. See https://www.drupal.org/core/d8-bc-policy for the `@internal` tag.